### DISCUSSION TITLE
Prisma reflection during runtime?

### DISCUSSION THREAD

**Discussion by raminnoodle** [prisma/prisma#12680] [Q&A]
Is it possible to determine if a table has a specific column during runtime through code?

**Comment by austincrim** [comment]
Hey @raminnoodle 👋 ,

What is your use case? That would impact potential solutions.

Is the table managed by the Prisma schema? If so, you should already know all columns at compile-time.

If the table isn't included in a Prisma schema, you could try doing a raw SQL query for the column and catching any errors thrown to determine if it wasn't found.

Let me know, thanks!

**Reply by raminnoodle** [reply]
Hi @austincrim!,

Yes all tables are fully managed by Prisma. 

Iv created a dynamic filter engine so users can build their own filters and store them as queues for my app. 
https://github.com/prisma/prisma/discussions/6801

The filter engine works across multiple types of tables. I am adding a feature now to enforce a type of record-level security. 

Table Examples:
```
WorkOrder {
  workerId
  teamId
  locationId
}

Equipment{
 locationId
}
```

The application has 3 primary types of users (Workers, Team Leads, & Location Admins).

When a worker views the queue I only want to filter to workorders where their workerid is associated.
When a teamlead views the queue I want them to see all workorders where their teamid is associated.
when a location admin views a queue I want them to see all workers in their location.

Some objects may contain all 3 levels of fields whereas others only have 1 or two.

I would really like to use reflection for when I'm dynamically building the queue filter I would look at the auth token and if the auth token has a workerid and the table has a worker id I would add that where clause. If they don't have a workerid and have a teamid and if the object has a teamid it would build the where clause for that. 

If there is no easy way to reflect back on a table to see its properties then I would most likely use the object.model.ts iv created to check if it contains a property.

Hope that makes sense. Sorry in advance I'm horrible at writing that's why I stick to programming.

**Reply by austincrim** [reply]
Thanks for the extra clarification! Sounds like an interesting project.

As far as runtime reflection, you might be able to use Prisma's own internal representation of its schema, the DMMF (data model meta format). 
```ts
import { Prisma } from '@prisma/client'

console.log(Prisma.dmmf)
```

The DMMF is an **unstable** API and is not subject to our usual semantic versioning strategy, so proceed with caution.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
